### PR TITLE
[Blaze] Stop showing Blaze in dashboard when site not eligible

### LINKED ISSUE #11782: Require WPCom login before directing to the campaign creation flow for users logged in without WPCom

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #11782

## Description

Fix a bug in which we displayed the Blaze view in the My Store tab (Dashboard) even though the current user is not eligible for Blaze. 

I added code to store the blaze eligibility check value and use the results from the storage layer only if the blaze eligibility check value is `true`. 

## Testing instructions
- Create a JN site with Woo but no Jetpack.
- Log in to the app using site credentials or application password.
- Switch to the Products tab, create a new product, and publish it.
- Switch to the My Store tab (Dashboard) and validate that the Blaze section is not displayed.
- Pull to refresh the dashboard and validate that the Blaze section is not displayed.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/5db6dc7f-0476-4e03-bc4a-5b383c5d0c86

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/182">17.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 17.1
 -----
+- [*] Blaze: Stop showing Blaze in dashboard when site not eligible. [https://github.com/woocommerce/woocommerce-ios/pull/11797]
```
Comment (itsmeichigo) [inline RELEASE-NOTES.txt]: Nit: Maybe we can reword this to make it clearer?

```suggestion
- [*] Blaze: Hide the Blaze section on the Dashboard screen when logged in without WPCom. [https://github.com/woocommerce/woocommerce-ios/pull/11797]
```

**Review by itsmeichigo — APPROVED** [review]
LGTM 🚢

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (selanthiraiyan) [reply RELEASE-NOTES.txt]: I agree. Done in https://github.com/woocommerce/woocommerce-ios/pull/11797/commits/c3269c29b010e42fe4404972c6080bc554e6d5af
